axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/GrpcBackedQueryMessage.java

```diff
@@ -133,4 +133,10 @@ public GrpcBackedQueryMessage<P, R> withMetaData(@Nonnull Map<String, String> me
     public GrpcBackedQueryMessage<P, R> andMetaData(@Nonnull Map<String, String> metaData) {
         return withMetaData(metaData().mergedWith(metaData));
     }
+
+    @Override
+    public <T> QueryMessage<T, R> withConvertedPayload(@Nonnull Type type, @Nonnull Converter converter) {
```

Comment:
What do you think about the `map` or `mapPayload` name? As we were discussing recently.

Reply:
Fair question! Let's think about it a little...

Reply:
Discussed in our weekly meetings that we'll stick to `withConvertedPayload`. 
Map or transform payload do seem nice, but suggest the use of a lambda, which we're definitely not doing here.
Lastly, using the "with..." wording flow aligns with our `andMetaData` and `withMetaData` methods.